torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,1063 @@
+    """
+
+    DOWNLOAD_SPACE = 400 * 1024 * 1024 * 1024
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "tree",
+        transforms: Optional[Callable] = None,
+        download: bool = False,
+        max_disparity: float = 256.0,
+    ):
+        super().__init__(root, transforms)
+
+        root = Path(root) / "CREStereo"
+        self.max_disparity = max_disparity
+
+        # if the API user requests a dataset download check that the user can download it
+        if download:
+            statvfs = os.statvfs(root)
+            # measured in bytes
+            available_space = statvfs.f_frsize * statvfs.f_bavail
+            if available_space - self.DOWNLOAD_SPACE < 0:
+                raise ValueError(
+                    f"The storage device for {root} is too small to download the dataset), "
+                    f"an additional {self.DOWNLOAD_SPACE - self.available_space:.2f} GB are required."
+                )
+            self._download_dataset(root)
+
+        verify_str_arg(split, "split", valid_values=("tree", "shapenet", "reflective", "hole", "all"))
+
+        splits = {
+            "tree": ["tree"],
+            "shapenet": ["shapenet"],
+            "reflective": ["reflective"],
+            "hole": ["hole"],
+            "all": ["hole", "shapenet", "reflective", "hole"],
+        }[split]
+
+        for s in splits:
+            imgs_left = sorted(glob(str(root / s / "*_left.jpg")))
+            imgs_right = list(p.replace("_left", "_right") for p in imgs_left)
+
+            if not len(imgs_left) or not len(imgs_right):
+                raise FileNotFoundError("No images found in {}".format(root))
+
+            imgs = list((left, right) for left, right in zip(imgs_left, imgs_right))
+            self._images += imgs
+
+            disparity_maps_left = list(p.replace("_left", "_left.disp") for p in imgs_left)
+            disparity_maps_right = list(p.replace("_right", "_right.disp") for p in imgs_right)
+
+            if not any(os.path.exists(file_path) for file_path in disparity_maps_left):
+                raise FileNotFoundError("No disparity valid maps found in {}".format(root / s))
+
+            if not any(os.path.exists(file_path) for file_path in disparity_maps_right):
+                raise FileNotFoundError("No disparity valid maps found in {}".format(root / s))
+
+            disparity_maps = list((left, right) for left, right in zip(disparity_maps_left, disparity_maps_right))
+            self._disparities += disparity_maps
+
+    def _read_disparity(self, file_path: str) -> Tuple:
+        disparity = np.array(Image.open(file_path), dtype=np.float32)
+        valid = (disparity < self.max_disparity) & (disparity > 0.0)
+        # unsqueeze the disparity map into (C, H, W) format
+        disparity = disparity[None, :, :]
+        return disparity, valid
+
+    def _download_dataset(self, root: str) -> None:
```

Comment:
For CREStereo at least it would be difficult to perform any checking given the dataset. We could at most verify and md5 for the entirity of the directory contents, however I might expect that to take long enough. I believe a viable alternative would be to check that the directory contains the right amount of files.

For Middleburry2014 I believe it would be much more simple however.